packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1208,21 +1207,21 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
 
   /// The line height at the start of the current selection.
   double get startGlyphHeight {
-    return _selectionDelegate.value.startSelectionPoint!.lineHeight;
+    return _selectionDelegate.value.startSelectionPoint?.lineHeight ?? 0;
```

Comment:
Nice catch, it looks like when either selection edge is not in the viewport then start/endSelectionPoint may be `null` which is causing the crash when scrolling the selection out of view..